routing/dht/records.go

```diff
@@ -100,14 +101,20 @@ func (dht *IpfsDHT) getPublicKeyFromNode(ctx context.Context, p peer.ID) (ci.Pub
 // key, we fail. we do not search the dht.
 func (dht *IpfsDHT) verifyRecordLocally(r *pb.Record) error {
 
-	// First, validate the signature
-	p := peer.ID(r.GetAuthor())
-	pk := dht.peerstore.PubKey(p)
-	if pk == nil {
-		return fmt.Errorf("do not have public key for %s", p)
+	if len(r.Signature) > 0 {
+		// First, validate the signature
+		p := peer.ID(r.GetAuthor())
+		pk := dht.peerstore.PubKey(p)
+		if pk == nil {
+			return fmt.Errorf("do not have public key for %s", p)
+		}
+
+		if err := record.CheckRecordSig(r, pk); err != nil {
+			return err
+		}
```

Comment:
this should still be part of the validators. see discussion on https://github.com/jbenet/go-ipfs/pull/802/files#r25189790